File: Reverse words in a given string - GFG/reverse-words-in-a-given-string.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends

class Solution
{
    public:
    //Function to reverse words in a given string.
    string reverseWords(string S) 
    { 
        // code here 
        stack<string> stk;
        string res ="";
        for(auto i: S){
            if(i != '.'){
                res += i;
            }else{
                stk.push(res);
                stk.push(".");
                res  = "";
            }
        }
        stk.push(res);
        res = "";
        while(!stk.empty()){
            res += stk.top();
            stk.pop();
        }
        
        return res;
    } 
};

//{ Driver Code Starts.
int main() 
{
    int t;
    cin >> t;
    while (t--) 
    {
        string s;
        cin >> s;
        Solution obj;
        cout<<obj.reverseWords(s)<<endl;
    }
}
// } Driver Code Ends

File: Reverse words in a given string - GFG/reverse-words-in-a-given-string.java
//{ Driver Code Starts
import java.util.*;
import java.lang.*;
import java.io.*;
class GFG {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t > 0) {
            String s = sc.next();
            Solution obj = new Solution();
            System.out.println(obj.reverseWords(s));
            t--;
        }
    }
}

// } Driver Code Ends



class Solution 
{
    //Function to reverse words in a given string.
    String reverseWords(String S)
    {
        // code here 
        
        ArrayList<String> arr = new ArrayList<>();
        StringBuilder sb = new StringBuilder();
        for(int i = 0;i<S.length();i++){
            if(S.charAt(i) == '.'){
                arr.add(sb.toString());
                sb.setLength(0);
                sb.append(S.charAt(i));
                arr.add(sb.toString());
                sb.setLength(0);
            }else{
                sb.append(S.charAt(i));
            }
        }
        arr.add(sb.toString());
        sb.setLength(0);
        
        Collections.reverse(arr);
        for(String i:arr){
            sb.append(i);
        }
        
        return sb.toString();
    }
}